### PR TITLE
feat: debian: Add guide to enable Wifi

### PR DESCRIPTION
**Pull request by jsuhaas22**
Support was recently added to Debian for OOB Wifi experience using CC33xx cards. So add documentation for it.

### CODE REVIEW

File: source/debian/Connecting_to_Wifi.rst
Comment (StaticRocket) [inline source/debian/Connecting_to_Wifi.rst:1]: This is not part of the Debian overview. It should be moved into a `How_to_Guides` sub-directory under `source/debian/`.

**Review by StaticRocket — DISMISSED** [review]
Why is this being documented like this? It's just reusing the majority of the `source/linux/How_to_Guides/Target/How_To_Enable_M2CC3301_in_linux.rst` entry with 1 new paragraph.

Why is this not just a landing page saying what version of Debian this feature was introduced in and then point people (via a ref) to the OS agnostic instructions already present in the Linux section for configuration an usage.

**Comment by jsuhaas22** [comment]
> Why is this being documented like this? It's just reusing the majority of the source/linux/How_to_Guides/Target/
> How_To_Enable_M2CC3301_in_linux.rst entry with 1 new paragraph.

> Why is this not just a landing page saying what version of Debian this feature was introduced in and then point people (via a ref) > to the OS agnostic instructions already present in the Linux section for configuration an usage.

@StaticRocket The instructions aren't entirely OS-agnostic. In debian, it is necessary to prepend `bash` to the scripts for running them.

And in the [Enable Overlay section](https://software-dl.ti.com/processor-sdk-linux/esd/AM62PX/10_00_07_04/exports/docs/linux/How_to_Guides/Target/How_To_Enable_M2CC3301_in_linux.html#enable-dt-overlay-for-m-2-cc33x1), there's a reference made to `tisdk-default-image`. Debian images instead contain the string `tisdk-debian-trixie`. This is probably a minor thing, but I thought to specify debian image by name explicitly.

**Comment by StaticRocket** [comment]
> > Why is this being documented like this? It's just reusing the majority of the source/linux/How_to_Guides/Target/
> > How_To_Enable_M2CC3301_in_linux.rst entry with 1 new paragraph.
> 
> > Why is this not just a landing page saying what version of Debian this feature was introduced in and then point people (via a ref) > to the OS agnostic instructions already present in the Linux section for configuration an usage.
> 
> @StaticRocket The instructions aren't entirely OS-agnostic. In debian, it is necessary to prepend `bash` to the scripts for running them.

Oh, cool. Then that is an upstream script bug. If they are using bashisms in the script then it should have a bash shebang regardless of what environment its running in.

> And in the [Enable Overlay section](https://software-dl.ti.com/processor-sdk-linux/esd/AM62PX/10_00_07_04/exports/docs/linux/How_to_Guides/Target/How_To_Enable_M2CC3301_in_linux.html#enable-dt-overlay-for-m-2-cc33x1), there's a reference made to `tisdk-default-image`. Debian images instead contain the string `tisdk-debian-trixie`. This is probably a minor thing, but I thought to specify debian image by name explicitly.

That's fair. But it could just as easily be an ifconfig section for `CONFIG_sdk in ['DebianSDK']`

File: source/debian/Connecting_to_Wifi.rst
Comment (jsuhaas22) [reply source/debian/Connecting_to_Wifi.rst:1]: Addressed this @StaticRocket.

It seems I had missed adding this guide to AM62PX, so did that too.

**Comment by jsuhaas22** [comment]
@StaticRocket 

> Oh, cool. Then that is an upstream script bug. If they are using bashisms in the script then it should have a bash shebang regardless of what environment its running in.

I'll report it to the maintainers

> That's fair. But it could just as easily be an ifconfig section for CONFIG_sdk in ('DebianSDK')

Oh I didn't know this is possible. Thanks! I've updated the PR.

File: source/linux/How_to_Guides/Target/How_To_Enable_M2CC3301_in_linux.rst [hunk 1]
```diff
@@ -74,12 +81,23 @@ Connect to Wi-Fi
...
+    .. code-block:: console
 
-    cd /usr/share/cc33xx
-    ./sta_start.sh
-    ./sta_connect.sh -s WPA-PSK -n <SSID> -p <PASSWORD>
-    udhcpc -i wlan0
+        cd /usr/share/cc33xx
+        bash ./sta_start.sh
+        bash ./sta_connect.sh -s WPA-PSK -n <SSID> -p <PASSWORD>
+        udhcpc -i wlan0
```
Comment (StaticRocket) [inline source/linux/How_to_Guides/Target/How_To_Enable_M2CC3301_in_linux.rst:100]: Can you split this change into a separate commit so it can be reverted when the upstream issue is addressed?

**Review by StaticRocket — CHANGES_REQUESTED** [review]

**Comment by sabeeh-khan** [comment]
For Debian, I imagine you still have to set the uEnv.txt to enable the m2 card? Are there any changes in that process?

**Comment by jsuhaas22** [comment]
@sabeeh-khan No change in that process. You still have to set `name_overlays` in the uEnv file or U-Boot prompt.

File: source/linux/How_to_Guides/Target/How_To_Enable_M2CC3301_in_linux.rst (hunk 1 above)
Comment (jsuhaas22) [reply source/linux/How_to_Guides/Target/How_To_Enable_M2CC3301_in_linux.rst:100]: Done, @StaticRocket

**Review by cshilwant — APPROVED** [review]
https://github.com/TexasInstruments/processor-sdk-doc/pull/72#issuecomment-2526811762 warnings can be ignored as they aren't introduced by this PR

**Review by StaticRocket — APPROVED** [review]

**Review by praneethbajjuri — APPROVED** [review]
